ksqldb-execution/src/main/java/io/confluent/ksql/execution/interpreter/TermCompiler.java

```diff
@@ -0,0 +1,559 @@
+import io.confluent.ksql.execution.expression.tree.SimpleCaseExpression;
+import io.confluent.ksql.execution.expression.tree.StringLiteral;
+import io.confluent.ksql.execution.expression.tree.SubscriptExpression;
+import io.confluent.ksql.execution.expression.tree.TimeLiteral;
+import io.confluent.ksql.execution.expression.tree.TimestampLiteral;
+import io.confluent.ksql.execution.expression.tree.Type;
+import io.confluent.ksql.execution.expression.tree.UnqualifiedColumnReferenceExp;
+import io.confluent.ksql.execution.expression.tree.WhenClause;
+import io.confluent.ksql.execution.interpreter.terms.BasicTerms;
+import io.confluent.ksql.execution.interpreter.terms.BasicTerms.BooleanTerm;
+import io.confluent.ksql.execution.interpreter.terms.ColumnReferenceTerm;
+import io.confluent.ksql.execution.interpreter.terms.CreateArrayTerm;
+import io.confluent.ksql.execution.interpreter.terms.CreateMapTerm;
+import io.confluent.ksql.execution.interpreter.terms.DereferenceTerm;
+import io.confluent.ksql.execution.interpreter.terms.FunctionCallTerm;
+import io.confluent.ksql.execution.interpreter.terms.IsNotNullTerm;
+import io.confluent.ksql.execution.interpreter.terms.IsNullTerm;
+import io.confluent.ksql.execution.interpreter.terms.LikeTerm;
+import io.confluent.ksql.execution.interpreter.terms.LogicalBinaryTerms;
+import io.confluent.ksql.execution.interpreter.terms.NotTerm;
+import io.confluent.ksql.execution.interpreter.terms.SearchedCaseTerm;
+import io.confluent.ksql.execution.interpreter.terms.StructTerm;
+import io.confluent.ksql.execution.interpreter.terms.SubscriptTerm;
+import io.confluent.ksql.execution.interpreter.terms.Term;
+import io.confluent.ksql.execution.util.CoercionUtil;
+import io.confluent.ksql.execution.util.ExpressionTypeManager;
+import io.confluent.ksql.function.FunctionRegistry;
+import io.confluent.ksql.function.GenericsUtil;
+import io.confluent.ksql.function.KsqlScalarFunction;
+import io.confluent.ksql.function.UdfFactory;
+import io.confluent.ksql.function.types.ArrayType;
+import io.confluent.ksql.function.types.ParamType;
+import io.confluent.ksql.function.types.ParamTypes;
+import io.confluent.ksql.function.udf.Kudf;
+import io.confluent.ksql.schema.ksql.Column;
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import io.confluent.ksql.schema.ksql.SchemaConverters;
+import io.confluent.ksql.schema.ksql.SqlArgument;
+import io.confluent.ksql.schema.ksql.types.SqlArray;
+import io.confluent.ksql.schema.ksql.types.SqlBaseType;
+import io.confluent.ksql.schema.ksql.types.SqlMap;
+import io.confluent.ksql.schema.ksql.types.SqlType;
+import io.confluent.ksql.util.DecimalUtil;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.KsqlException;
+import io.confluent.ksql.util.Pair;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@SuppressWarnings("checkstyle:ClassDataAbstractionCoupling")
```

Comment:
nit: we've historically been using comment style suppressions for checkstyle instead of the annotation

Reply:
Changed.